apps/studio/components/interfaces/Auth/Users/UsersList.tsx

```diff
@@ -36,59 +37,64 @@ const UsersList = ({
-            <Table.th className="table-cell">Last Sign In</Table.th>
-            <Table.th className="table-cell">User UID</Table.th>
-            <Table.th></Table.th>
-          </>
-        }
-        body={
-          <>
-            {isSuccess && users.length == 0 && (
-              <Table.tr>
-                {/* @ts-ignore */}
-                <Table.td
-                  colSpan={8}
-                  className="h-14 whitespace-nowrap border-t p-4 text-sm leading-5 text-gray-300"
-                >
-                  <div className="flex items-center space-x-3 opacity-75">
-                    <IconAlertCircle size={16} strokeWidth={2} />
-                    <p className="text-foreground-light">
-                      {keywords
-                        ? `No users matched the search query "${keywords}"`
-                        : 'No users in your project yet'}
-                    </p>
-                  </div>
-                </Table.td>
-              </Table.tr>
-            )}
-            {isSuccess &&
-              users.length > 0 &&
-              users.map((x: any) => (
-                <UserListItem
-                  key={x.id}
-                  user={x}
-                  canRemoveUser={canRemoveUser}
-                  canRemoveMFAFactors={canRemoveMFAFactors}
-                />
-              ))}
+    <Table
+      head={
+        <>
+          <Table.th>Display Name</Table.th>
+          <Table.th>Email</Table.th>
+          <Table.th>Phone</Table.th>
+          <Table.th className="table-cell">Provider</Table.th>
+          <Table.th className="table-cell">Created</Table.th>
+          <Table.th className="table-cell">Last Sign In</Table.th>
+          <Table.th className="table-cell">User UID</Table.th>
+          <Table.th></Table.th>
+        </>
+      }
+      body={
+        <>
+          {isLoading && (
+            <>
+              <UsersListItemSkeleton />
+              <UsersListItemSkeleton />
+              <UsersListItemSkeleton />
+              <UsersListItemSkeleton />
+            </>
+          )}
+          {isSuccess && users.length == 0 && (
             <Table.tr>
-              <Table.td colSpan={8}>
-                <UsersPagination total={total} page={page} setPage={setPage} />
+              <Table.td
+                colSpan={8}
+                className="h-14 whitespace-nowrap border-t p-4 text-sm leading-5 text-gray-300"
+              >
+                <div className="flex items-center space-x-3 opacity-75">
+                  <IconAlertCircle size={16} strokeWidth={2} />
+                  <p className="text-foreground-light">
+                    {keywords
+                      ? `No users matched the search query "${keywords}"`
+                      : 'No users in your project'}
+                  </p>
+                </div>
               </Table.td>
             </Table.tr>
-          </>
-        }
-      />
-    </Loading>
+          )}
+          {isSuccess &&
```

Comment:
while we're refactoring this, probably add isError handler? with `<AlertError/>`

Reply:
Good shout! I've added it purely based on `error` since `isError` would be passed as a prop so TS wouldn't be any extra help